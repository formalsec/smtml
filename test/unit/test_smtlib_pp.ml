open Smtml

let () =
  let x = Expr.symbol @@ Symbol.make Ty_int "x" in
  let y = Expr.symbol @@ Symbol.make Ty_int "y" in
  let script =
    Ast.Script.of_exprs
      [ Expr.relop Ty_int Eq
          (Expr.binop Ty_int Add x y)
          (Expr.value (Value.Int 30))
      ; Expr.relop Ty_int Gt x
          (Expr.binop Ty_int Mul (Expr.value (Value.Int 2)) y)
      ; Expr.relop Ty_int Gt y (Expr.value (Value.Int 10))
      ]
  in
  Fmt.pr "@[<v>%a@]@." Ast.Script.pp script
