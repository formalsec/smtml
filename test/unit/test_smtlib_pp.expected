(declare-const x Int)
(declare-const y Int)
(assert (= (+ x y) 30))
(assert (< (* 2 y) x))
(assert (< 10 y))
(check-sat)
